File: test/src/around_en.py
# General solution to worlds Around 1 to 4
think(0)

def turn_right():
    repeat 3:
        turn_left()

def follow_right_wall():
    if right_is_clear():
        turn_right()
        move()
    elif front_is_clear():
        move()
    else:
        turn_left()

put()
while wall_in_front():
    turn_left()
move()

while not object_here():
    follow_right_wall()

File: tests/integration_tests/tutorial_en/programs/around_en.js
// overkill - single program for Around 1, 2, 3, 4
function right() {
    turn_left();
    turn_left();
    turn_left();
}

function follow_wall () {
    if (right_is_clear()) {
        right();
        move();
    } else if (front_is_clear()) {
        move();
    } else {
        turn_left();
    }
}

put();
while(!front_is_clear()) {
    turn_left();
}
move();

while (!object_here()) {
    follow_wall();
}
